gp-nested-forms/gpnf-auto-attach-child-files.php

```diff
@@ -11,68 +11,134 @@
  * Plugin URI:   http://gravitywiz.com/documentation/gravity-forms-nested-forms/
  * Description:  Auto-attach Uploaded Files from Child to Parent Notifications
  * Author:       Gravity Wiz
- * Version:      0.1
+ * Version:      0.2
  * Author URI:   https://gravitywiz.com/
  */
-add_filter( 'gform_notification', function( $notification, $form, $entry ) {
-	// Notes:
-	// - If notifications IDs are specified this snippet will only work for those specific forms.
-	// - If the parent form has upload fields, they will be attached instead of the child form's fields.
-	// - If neither of these conditions are present, child form uploads will be attached by default.
-	// Configuration:
-	// 1) Ensure that the parent form's notification has the "Attach uploaded fields to notification" checked.
-	// 2) [Optional] Modify the following array to limit the snippet to specific notification IDs.
-	// Notification IDs can be retrieved from the nid parameter in the URL when editing a notification.
-	// Example: $notification_ids = array( '5daaedb49dc32', '5dbce25cc21c2' );
-	$notification_ids = array();
-
-	if ( ! class_exists( 'GPNF_Entry' ) ) {
-		return $notification;
+class GPNF_Auto_Attach_Child_Files {
+	private $notification_ids = array();
+
+	private $child_form_upload_field_ids = array();
+
+	function __construct( $config = array() ) {
+		$this->set_config_data( $config );
+		$this->init_hooks();
 	}
 
-	$upload_fields = GFCommon::get_fields_by_type( $form, array( 'fileupload' ) );
+	function init_hooks() {
+		add_filter( 'gform_notification', array( $this, 'handle_notification_attachments' ), 10, 3 );
+	}
 
-	if ( ! empty( $notification_ids ) && ! in_array( $notification['id'], $notification_ids ) ) {
-		return $notification;
-	} else {
-		// If parent form has upload fields, rely on the notification's Attachments setting.
-		if ( ! empty( $upload_fields ) ) {
-			if ( ! rgar( $notification, 'enableAttachments', false ) ) {
-				return $notification;
-			}
+	function set_config_data( $config ) {
+		if ( array_key_exists( 'notification_ids', $config ) ) {
```

Comment:
FWIW, `array_key_exists( 'notification_ids', $config )` can be safely (no PHP notices) shorted to one of the following:

```php
if ( isset( $config['notification_ids'] ) ) {
```

or

```php
if ( ! empty( $config['notification_ids'] ) ) {
```

or even shorter (no if statement)...

```php
$this->notification_ids = rgar( $config, 'notification_ids', array() );
```

Reply:
Noice!

Unfortunately `rgar` isn't always present since this seems to be instantiated before Gravity Forms but I think one of `isset` or `empty` will work :+1:

Reply:
Done

Reply:
Good call there! If you ever do need GF functions (or functions from perks), you can hook a method to the `plugins_loaded` action.